### ISSUE TITLE
[Port] Why `G.weights` is the transpose of `adjacency_matrix(G)`

### ISSUE THREAD

**Issue by gdalle** [JuliaGraphs/SimpleWeightedGraphs.jl#11]
See https://github.com/sbromberger/SimpleWeightedGraphs.jl/issues/65

**Comment by hdavid16** [comment]
I think this needs to be opened up to discussion as this conflicts with the `get_weight` function, which is not using the transpose. As an example:

```julia
julia> s = SimpleWeightedDiGraph([1,2,1], [2,1,2], [1,1,1])
{2, 2} directed simple Int64 graph with Int64 weights     

julia> get_weight(s,1,2)
2
```
Which is wrong because the weight of the `Edge 1 => 2` is 1, not 2. This occurs because, `get_weight` does not use the transpose of the graph weights.

```julia
get_weight(g::AbstractSimpleWeightedGraph, u::Integer, v::Integer) = g.weights[v, u]
```

I disagree with the explanation in the referenced issue on the old repository (https://github.com/sbromberger/SimpleWeightedGraphs.jl/issues/65#issuecomment-751356453). Why would you be more interested in `outneighbors` than `inneighbors`? In most of my applications where I use Graphs (supply chain networks and process networks), I am just as interested in the incoming edges as in the outgoing edges of a vertex. I don't think you should transpose the matrix just for performance sake as this can be very confusing to the user. 

My personal opinion is that we should not be using the transpose. However if the community prefers to use the transpose then we have a bug in `get_weight`.

**Comment by hdavid16** [comment]
Nevermind, I see that the issue is referenced is in calling `G.weights` instead of `weights(G)`. Internally, the weight vector is stored as the transpose (for performance issues). But the user should never see the transpose if they are using the API properly (i.e., `weights(G)` instead of accessing the internals `G.weights`).

**Comment by gdalle** [comment]
The reason this issue was still open in the original repo is because many algorithms used `g.weights` instead of `weights(g)`, which would be the more julian way. This has been fixed by #14
